### PR TITLE
CR-1153424 IOMMU Status dump

### PR DESCRIPTION
**Pull request by prapulkrishnamurthy**
#### Problem solved by the commit
Iommu status check from iommu grub logs

#### How problem was solved, alternative solutions (if any) and why they were rejected
Parsed iommu event logs from Dmesg and dumped into service bundle output directory.

#### What has been tested and how, request additional testing if necessary
Tested on VCK5000 and found the output at dump files in output directory.
image

### CODE REVIEW

File: src/runtime_src/tools/scripts/service_bundle.sh [hunk 1]
```diff
@@ -58,6 +58,12 @@ done
 # Funcs start here
 #############################
```
Comment (xdavidz) [inline src/runtime_src/tools/scripts/service_bundle.sh:60]: this is good, but please add the following script that I attached in the CR's description
```
for g in `find /sys/kernel/iommu_groups/* -maxdepth 0 -type d | sort -V`; do
    echo "IOMMU Group ${g##*/}:"
    for d in $g/devices/*; do
        echo -e "\t$(lspci -nns ${d##*/})"
    done;
done;
```

make this as a function and then dump the info into a separate file
field engineer is using this script.

Comment (prapulkrishnamurthy) [reply src/runtime_src/tools/scripts/service_bundle.sh:60]: Added changes as per comments.

File: src/runtime_src/tools/scripts/service_bundle.sh [hunk 2]
```diff
@@ -182,6 +204,8 @@ funcArray=(
 	vmr_logs
 	debug_logs
 	core_dumps
+	iommu_dump
+	iommu_groups
 )
```
Comment (xdavidz) [inline src/runtime_src/tools/scripts/service_bundle.sh:209]: can you please dump the iommu dump and iommu groups into host_info as iommu_dump.log and iommu_groups.log?

Comment (xdavidz) [reply src/runtime_src/tools/scripts/service_bundle.sh:209]: either 2 files or 1 file combined with two logs are ok

**Review by xdavidz — APPROVED** [review]

**Comment by xdavidz** [comment]
@prapulkrishnamurthy this is approved by me last week. Please ping all stockholders for any other potential concerns or seek for approval. Then, you can ask Max, Vamshi or those who can merge your PR. Note: the PR owner is responsible for moving your PR forward.

**Review by rozumx — APPROVED** [review]
Looks good.

I do find it interesting that this script doesn't check to see if the output directory exists.   If in the future you find yourself updating the script, you might wish to look adding this EOU feature.    

Doing so with most likely result in the need of adding a "force" or a "yes" option to create if that situation should arise.

**Comment by prapulkrishnamurthy** [comment]
> Looks good.
> 
> I do find it interesting that this script doesn't check to see if the output directory exists. If in the future you find yourself updating the script, you might wish to look adding this EOU feature.
> 
> Doing so with most likely result in the need of adding a "force" or a "yes" option to create if that situation should arise.

@rozumx thanks for the input and that sounds like a good improvement. But I don't think we need that because the script is designed to dump different logs and logs are allocated for different subfolders within a root folder; as simple as that. But adding more checks like this will complicate the ease of use (For field tester let's say) and we don't have to over-simplify it.

**Pull request merged**